sign.sh

```diff
@@ -52,35 +62,37 @@ checkSignConfiguration() {
 # Sign the built binary
 signRelease()
 {
+  TIMESTAMPSERVERS=`cut -d= -f2 < $WORKSPACE/$TIMESTAMP_SERVER_CONFIG | tr -d \\\\r`
+
   case "$OPERATING_SYSTEM" in
     "windows")
       echo "Signing Windows release"
       signToolPath=${signToolPath:-"/cygdrive/c/Program Files (x86)/Windows Kits/10/bin/10.0.17763.0/x64/signtool.exe"}
 
       # Sign .exe files
-      FILES=$(find . -type f -name '*.exe')
-      echo "$FILES" | while read -r f;
-      do
-        echo "Signing ${f}"
-        if ! "$signToolPath" sign /f "${SIGNING_CERTIFICATE}" /p "$SIGN_PASSWORD" /fd SHA256 /t http://timestamp.globalsign.com/scripts/timestamp.dll "$f"; then
-          echo "RETRYWARNING: Failed to sign ${f} at $(date +%T): Possible timestamp server error - RC $? ... Retrying in 10 seconds"
-          sleep 10s
-          "$signToolPath" sign /f "${SIGNING_CERTIFICATE}" /p "$SIGN_PASSWORD" /fd SHA256 /t http://timestamp.globalsign.com/scripts/timestamp.dll "$f"
-        fi        
-      done
-
-      # Sign .dll files
-      FILES=$(find . -type f -name '*.dll')
-      echo "$FILES" | while read -r f;
+      FILES=$(find . -type f -name '*.exe' -o -name '*.dll')
+      for f in $FILES
       do
         echo "Signing ${f}"
-        if ! "$signToolPath" sign /f "${SIGNING_CERTIFICATE}" /p "$SIGN_PASSWORD" /fd SHA256 /t http://timestamp.globalsign.com/scripts/timestamp.dll "$f"; then
-          echo "RETRYWARNING: Failed to sign ${f} at $(date +%T): Possible timestamp server error - RC $? ... Retrying in 10 seconds"
-          sleep 10s
-          "$signToolPath" sign /f "${SIGNING_CERTIFICATE}" /p "$SIGN_PASSWORD" /fd SHA256 /t http://timestamp.globalsign.com/scripts/timestamp.dll "$f"
+        STAMPED=false
+        for SERVER in $TIMESTAMPSERVERS; do
+          if [ "$STAMPED" = "false" ]; then
```

Comment:
Does that need to be `==`? and two `[[`?

Reply:
Simple answer: No

Long answer: What I've written is the correct POSIX-compliant version for a basic string compare without specific bash-isms, therefore none of those are necessary

Reply:
Simple answer: No

What I've written is the correct POSIX-compliant version without specific bash-isms